optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/common/JoinerType.java

```diff
@@ -41,4 +51,34 @@ public JoinerType flip() {
         }
     }
 
+    public boolean matches(Object left, Object right) {
+        return matcher.test(left, right);
+    }
+
+    private static boolean lessThan(Comparable left, Object right) {
+        return left.compareTo(right) < 0;
+    }
+
+    private static boolean lessThanOrEqual(Comparable left, Object right) {
+        return left.compareTo(right) <= 0;
+    }
+
+    private static boolean greaterThan(Comparable left, Object right) {
+        return left.compareTo(right) > 0;
+    }
+
+    private static boolean greaterThanOrEqual(Comparable left, Object right) {
+        return left.compareTo(right) >= 0;
+    }
+
+    private static boolean disjointColections(Collection leftCollection, Collection rightCollection) {
```

Comment:
if the enum is called "DISJOINT", this method should be called "disjointColections(...)" for consistency, not "disjointColections(...)".